### ISSUE TITLE
Wrong gatling report when using groups due to unknown delay before redirect starts

### ISSUE THREAD

**Issue by rohitmpr** [gatling/gatling#3385]
Hi,
We are using latest version of gatling 2.3.0 with nested group. While analysing gatling report generated by gatling-maven-plugin we saw discrepancy in **mean** value at group level are its childs. Some of the group shown very last different in child's total compared with parent group.

So for investigation we executed load test with less requests like
* 2 users making 1 request each  ==> Total 2 requests
* 1 user making 2 requests  ==> Total 2 requests

We are aware about group timings are calculated as per [this doc](https://gatling.io/docs/current/general/timings/#cumulated-time)

_Cumulated Time
The cumulated time of a group is the elapsed time from the start of the group’s first request to the end of the group’s last request, including pauses. Groups’ cumulated time are only reported in the “Cumulated response time” chart._

Here is our finding which seems issue
**Start time of every request which is redirected is with added delay, somewhere it's 10ms and somewhere is less**

Please find below simulation.log file and report for your reference.
**2 users 1 request each**
https://pastebin.com/SenCg4XT

Please note difference between logout.html end time and logout.html redirect start time
* First user (1511204698964) - (1511204698975) >> 11ms
* Second user (1511204702053) - (1511204702056) >> 3ms

![image](https://user-images.githubusercontent.com/10458705/33036810-92e2a0d0-cdfd-11e7-8fda-37461ce74b89.png)
In this image, mean time of expanded group SSO2_6_InstLogOut is wrong.

**1 user with total 2 request**

https://pastebin.com/zenqBzfq

Please note difference between logout.html end time and logout.html redirect start time
* First request (1511205793238) - (1511205793247) >> 11ms
* Second request (1511205795597) - (1511205795599) >> 2ms

![image](https://user-images.githubusercontent.com/10458705/33037135-95b253ea-cdfe-11e7-8b1b-a9146b33458c.png)
In this image, mean time of expanded group SSO2_6_InstLogOut is wrong.

Calculation is correct as per the doc mentioned [here](https://gatling.io/docs/current/general/timings/#cumulated-time) but gap before starting redirect request is still questionable and does not explain why it's there.

Note: Our code base is on company's intranet and not sharable, so unfortunately do don't have code snippet which I can share. But I can try to create demo project incase above mentioned details are not sufficient.

**Comment by slandelle** [comment]
> But I can try to create demo project incase above mentioned details are not sufficient.

Yes, that would be great.

Also, what is your OS?

**Comment by rohitmpr** [comment]
RHEL & OSX, on both I was able to reproduce it

**Comment by rohitmpr** [comment]
I am working on a demo project and just found out, while using example class from https://gatling.io/docs/current/quickstart/#gatling-scenario-explained. Issue is still reproducible even without any grouping

`RUN	com.jarvis.gatling.simulation.BasicSimulation		basicsimulation	1511208914425	null	2.0
USER	BasicSimulation	1	START	1511208916023	1511208916023
REQUEST	BasicSimulation	1		request_1	1511208916049	1511208917736	OK	 
REQUEST	BasicSimulation	1		request_1 Redirect 1	1511208917747	1511208917874	OK	 
USER	BasicSimulation	1	END	1511208916023	1511208922889`

**Comment by vkadam** [comment]
Sorry switching github account. 

Here is **code snippet** to reproduce issue

```
package com.jarvis.gatling.simulation

import io.gatling.core.Predef._
import io.gatling.core.structure.ChainBuilder
import io.gatling.http.Predef._

class BasicSimulation extends Simulation {

  val httpConf = http
    .baseURL("http://computer-database.gatling.io")
    .acceptHeader("text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
    .doNotTrackHeader("1")
    .acceptLanguageHeader("en-US,en;q=0.5")
    .acceptEncodingHeader("gzip, deflate")
    .userAgentHeader("Mozilla/5.0 (Windows NT 5.1; rv:31.0) Gecko/20100101 Firefox/31.0")

  val scn = scenario("BasicSimulation")
    .exec(getGroup("Group 1", makeHttpRequest))

  private def getGroup(groupName: String, chain: ChainBuilder) = {
    group(groupName)(chain)
  }

  private def makeHttpRequest = {
    exec(http("request_1").get("/"))
  }

  setUp(scn.inject(atOnceUsers(1))
  ).protocols(httpConf)
}
```

**simulation.log** generate from above simulation

```
RUN	com.jarvis.gatling.simulation.BasicSimulation		basicsimulation	1511210788722	null	2.0
USER	BasicSimulation	1	START	1511210790281	1511210790281
REQUEST	BasicSimulation	1	Group 1	request_1	1511210790308	1511210791608	OK	 
REQUEST	BasicSimulation	1	Group 1	request_1 Redirect 1	1511210791621	1511210791746	OK	 
GROUP	BasicSimulation	1	Group 1	1511210790286	1511210791756	2725	OK
USER	BasicSimulation	1	END	1511210790281	1511210791758
```
Noticeable difference between 1511210791608 & 1511210791621

**Gatling report**
![image](https://user-images.githubusercontent.com/972185/33040855-5e34e08e-ce0a-11e7-89f5-88f042a10763.png)

**Comment by vkadam** [comment]
I can understand it won't be possible to fix each and every possible path. We will be ok it as long as that timing is accounted across either of child request.

**Comment by slandelle** [comment]
> We will be ok it as long as that timing is accounted across either of child request.

No, that won't happen. We want to report HTTP response time and don't add processing time overhead. Once again this overhead doesn't occur under load but only for the first virtual user. And we're talking about some tens of milliseconds.

**Comment by vkadam** [comment]
Could this be related to #3386?

**Comment by slandelle** [comment]
2 different things:
* group cumulated response time was wrong when we had a redirect => fixed in #3386 
* group duration accounts for pause + processing time, and processing time can be as long as 50ms for the first user(s) => won't fixed
